src/fdc/MSXtar.cc

```diff
@@ -186,10 +188,12 @@ MSXtar::~MSXtar()
 	if (!fatCacheDirty) return;
 
 	for (auto i : xrange(sectorsPerFat)) {
-		try {
-			disk.writeSector(i + fatStart, fatBuffer[i]);
-		} catch (MSXException&) {
-			// nothing
+		for (auto fat : xrange(fatCount)) {
+			try {
+				disk.writeSector(i + fatStart + fat * sectorsPerFat, fatBuffer[i]);
+			} catch (MSXException&) {
+				// nothing
+			}
 		}
```

Comment:
Just checking. You unconditionally overwrite the 2nd FAT with the 1st FAT. Internally we only keep and update a single FAT copy. Right?

(Detail 1) Also here you could swap the order of the inner and outer loops.
(Detail 2) We could use the SectorAccessibleDisk::writeSectors() function to write all sectors (of a single FAT) at once. (Was already possible before your change).

Reply:
Yes, it keeps the two in sync. But if the FAT has never been written to (only read operations like `diskmanipulator export`) then it won’t write them.

Reply:
Re. detail 2, how do I convert a MemBuffer to a span?

Reply:
`MemBuffer` is very lightweight, it doesn't know the size of the buffer it owns. You can think of `std::span` as the combination of a (non-owning) pointer and a size. So you need to construct a `span` by combining the pointer obtained from `fatBuffer` with `sectorsPerFat`. Maybe something like this (not tested): `std::span{fatBuffer.data(), sectorsPerFat}`